### DISCUSSION TITLE
How do you send HTTP requests (and other logs) to Loki -- without using `push`

### DISCUSSION THREAD

**Discussion by stevenG-dv** [grafana/faro-web-sdk#264] [Q&A] [answered]
I have the set up below. It is definitely working as I am seeing my Trace-Ids in Tempo, but I am not getting any logs whatsoever in Loki. 

```
initialize() {
    const FARO_URL = ...;
    const NAME = '...';
    const VERSION = '...';

    initializeFaro({
      url: ...,
      apiKey: '...',
      app: {
        name: ...,
        version: '...',
        environment: 'env',
      },
      instrumentations: [
        new ConsoleInstrumentation({
          disabledLevels: [LogLevel.INFO, LogLevel.LOG, LogLevel.TRACE, LogLevel.DEBUG],
        }),
        new SessionInstrumentation(),
        new WebVitalsInstrumentation(),
      ]
    });

    faro.api.initOTEL(trace, context);

    const resource = Resource.default().merge(
      new Resource({
        [SemanticResourceAttributes.SERVICE_NAME]: NAME,
        [SemanticResourceAttributes.SERVICE_VERSION]: VERSION,
      })
    );

    const provider = new WebTracerProvider({ resource });

    provider.addSpanProcessor(new FaroSessionSpanProcessor(new BatchSpanProcessor(new FaroTraceExporter({ ...faro })), faro.metas));

    provider.register({
      propagator: new W3CTraceContextPropagator(),
      contextManager: new ZoneContextManager(),
    });

    const ignoreUrls = [FARO_URL];
    registerInstrumentations({
      instrumentations: [
        new XMLHttpRequestInstrumentation({ ignoreUrls }),
        new FetchInstrumentation({ ignoreUrls }),
      ],
    });
  }

**Comment by codecapitano** [comment, marked as answer]
Hi @stevenG-dv 
The Otel integration only collects (otel) traces.
If you want to collect specific logs the only way atm is to use the respective Faro APIs.

**Reply by stevenG-dv** [reply]
Thanks! I kind of suspected this -- could you possibly reference specifically which ones pass the data to Loki? (more importantly, just http calls/responses)

**Reply by codecapitano** [reply]
You mean which Faro APIs do send logs?

Generally speaking everything besides Traces is send as a Log.

These are the APIs which send logs:
* The logs API: `pushLog()`
* The exceptions API: `pushError()`
* The events API: `pushEvent()`
* The measurements API: `pushMeasurement()`

If you refer to instrumentations which track http calls :
* The [experimental fetch instrumentation](https://github.com/grafana/faro-web-sdk/tree/main/experimental/instrumentation-fetch) sends events to Loki.

**Reply by stevenG-dv** [reply]
Thank you. I actually added the faro FetchInstrumentation this morning -- but still not seeing any logs. Perhaps I have to also use the `fetch` methods? But I assume this means it needs to be called each time you want to log something. 
```
import { FetchInstrumentation} from '@grafana/faro-instrumentation-fetch';
...
    initializeFaro({
      url: ...,
      apiKey: '...',
      app: {
        ...
      },
      instrumentations: [
        new ConsoleInstrumentation({
           ...
        }),
        new FetchInstrumentation(),
      ]
    });

**Reply by codecapitano** [reply]
Exactly, the Fetch Instrumentation only instruments the Fetch API.

Note:
It may clash if you add the respective Otel instrumentations at the same time!

**Reply by codecapitano** [reply]
Implementation of the `XmlHttpRequest` API is WIP and not merged yet.

See:
https://github.com/grafana/faro-web-sdk/pull/239
